DALOptimizer/CSharpProject.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using ICSharpCode.NRefactory.CSharp;
+using ICSharpCode.NRefactory.TypeSystem;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Logging;
+
+namespace DALOptimizer
+{
+	/// <summary>
+	/// Represents a C# project (.csproj file)
+	/// </summary>
+	public class CSharpProject
+	{
+		/// <summary>
+		/// Parent solution.
+		/// </summary>
+		public readonly Solution Solution;
+		
+		/// <summary>
+		/// Title is the project name as specified in the .sln file.
+		/// </summary>
+		public readonly string Title;
+		
+		/// <summary>
+		/// Name of the output assembly.
+		/// </summary>
+		public readonly string AssemblyName;
+		
+		/// <summary>
+		/// Full path to the .csproj file.
+		/// </summary>
+		public readonly string FileName;
+		
+		public readonly List<CSharpFile> Files = new List<CSharpFile>();
+		
+		public readonly CompilerSettings CompilerSettings = new CompilerSettings();
+		
+		/// <summary>
+		/// The unresolved type system for this project.
+		/// </summary>
+		public readonly IProjectContent ProjectContent;
+		
+		/// <summary>
+		/// The resolved type system for this project.
+		/// This field is initialized once all projects have been loaded (in Solution constructor).
+		/// </summary>
+		public ICompilation Compilation;
```

Comment:
@sushantyelpale rename all variable names in camel case with first letter in lower case.